src/booking/views.py

```diff
@@ -0,0 +1,30 @@
+from .models import Book
+from .serializers import BookingSerializer
+from rest_framework import generics, status
+from rest_framework.response import Response
+from middlewares.permissions import IsOwner, IsOwnerOrPostOnly
+
+
+class BookList(generics.ListCreateAPIView):
+    serializer_class = BookingSerializer
+    permission_classes = [IsOwnerOrPostOnly]
+
+    def perform_create(self, serializer):
+        serializer.save(createdBy=self.request.user)
+
+    def list(self, request):
+        queryset = Book.objects.filter(createdBy=request.user)
+        page = self.paginate_queryset(queryset)
+
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+
+        serializer = BookingSerializer(queryset, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class BookDetail(generics.RetrieveUpdateAPIView):
+    queryset = Book.objects.all()
```

Comment:
class là bookdetail xong xuống Book...all() không hiểu  ???

Reply:
Tham khảo link này nhé https://www.django-rest-framework.org/tutorial/3-class-based-views/

Reply:
đúng rồi, details mà sao lại là all?

Reply:
?

Reply:
Dạ vì class BookDetail kế thừa từ RetrieveUpdateAPIView. Theo em nghĩ là khi chương trình bắt đầu chạy thì sẽ query toàn bộ dữ liệu từ database và lưu vào bộ nhớ của máy. Nó sẽ nhanh hơn là mỗi lần muốn lấy dữ liệu phải query tới database

Reply:
thế e dùng hàm detail làm gì? sao ko dùng hàm list thôi ?

Reply:
e dùng class này để update nữa mà a

Reply:
update là update 1 hay nhiều?

Reply:
mỗi lần gọi đến function này là nó có load lại cái queryset kia ko? hay là nếu e lấy all rồi thì ko load lại nữa?

Reply:
update một thôi ạ. query đó chỉ chạy một lần lúc chạy chương trình thôi ạ